llvm/lib/IR/Verifier.cpp

```diff
@@ -5008,6 +5008,9 @@ void Verifier::visitProfMetadata(Instruction &I, MDNode *MD) {
       Check(mdconst::dyn_extract<ConstantInt>(MDO),
             "!prof brunch_weights operand is not a const int");
     }
+  } else {
+    Check(ProfName == "VP", "expected either branch_weights or VP profile name",
```

Comment:
The PR description says that the validation "moved" - did this exist somewhere else already?

Reply:
oh. you're right, it didn't "move", I more thought of "it used to be checked by users <see tests I changed>, but now it's in the verifier". changing title.